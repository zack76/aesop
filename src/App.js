import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getProducts } from './reducers/product/getProducts';
import ItemService from './service/ItemService'
import NavBar from './components/NavBar'

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            text: ''
        }
    }

    componentDidMount() {
        ItemService.getItemDetail()
            .then(res => res.json())
            .then(resp => {
                console.log(resp)
                this.props.getProducts(resp)
            })
    }


    render() {
        return (
            <div className="App">
                <div className="container">
                    <NavBar />
                </div>
            </div>
        );
    }
}

const mapStateToProps = ({ products }) => {
    return {
        products: products
    }
}

export default connect(mapStateToProps , { getProducts })(App);
