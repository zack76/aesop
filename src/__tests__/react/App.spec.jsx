import React from 'react';
import { shallow } from 'enzyme';
import App from '../../App';

describe('App', () => {
  it('should render without error', () => {
    const wrapper = shallow(<App debug />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a Home component inside App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('Connect(Home)').length).toBe(1);
  });
});
