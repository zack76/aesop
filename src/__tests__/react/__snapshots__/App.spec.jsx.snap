// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without error 1`] = `<Connect(Home) />`;
